### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by bgamari**
Here are a variety of changes I've had in my local tree:

* Fix Haddock PDF documentation build
* Ensure that we add all system library dependencies to `LD_LIBRARY_PATH`
* Add DTrace support

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -18,8 +18,9 @@ in
 , withDocs  ? true
 , withIde   ? false
 , withHadrianDeps ? false
-, withDwarf ? nixpkgs.stdenv.isLinux  # enable libdw unwinding support
-, withNuma  ? nixpkgs.stdenv.isLinux
+, withDwarf  ? nixpkgs.stdenv.isLinux  # enable libdw unwinding support
+, withNuma   ? nixpkgs.stdenv.isLinux
+, withDtrace ? nixpkgs.stdenv.isLinux
```
Comment (Ericson2314) [inline default.nix]: Not darwin?

Comment (bgamari) [reply default.nix]: Nope, I have no means of testing Darwin. If someone else wants to figure our the Darwin situation they should feel free ;)

Comment (supersven) [reply default.nix]: At least `linuxPackages.systemtap` isn't available under Darwin.

I've never used DTrace for GHC, so I don't know what's really needed. But if you've something to test, I can easily start a `nix-shell` and try it out... 😄

Comment (supersven) [reply default.nix]: BTW, there is `darwin.dtrace` in nixpkgs and it looks like Darwin is shipped with `dtrace` (`/usr/sbin/dtrace`).

Comment (bgamari) [reply default.nix]: To be honest, I'm somewhat suspect that GHC's DTrace support has bit-rotted. Nevertheless, it does build with this change. You don't need to do anything in particular; `configure` automatically picks up that the `dtrace` executable is available.

**Pull request merged**